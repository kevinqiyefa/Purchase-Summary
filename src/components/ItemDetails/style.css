.Item-Details-Main {
  width: 15em;
}

#add::after {
  content: '\002B';
}

#minus::after {
  content: '\2212';
}

.Item-Details-Title {
  display: flex;
  align-items: center;
  cursor: pointer;
}

.Item-Details-Main h4 {
  text-decoration: underline;
}

#add,
#minus {
  padding-left: 0.5em;
  font-size: 2em;
  padding-bottom: 0.25em;
}

.item-details {
  display: flex;
  margin-bottom: 1em;
}

#item-name {
  font-size: 0.8em;
}

#item-qty {
  display: flex;
  justify-content: space-between;
  font-size: 0.9em;
  margin-bottom: 0.5em;
}

#item-picture img {
  width: 5em;
  padding-top: 0.5em;
}
.item-saving-prices {
  display: flex;
  flex-direction: column;
}

.item-subtotal {
  color: gray;
  text-decoration: line-through;
}
