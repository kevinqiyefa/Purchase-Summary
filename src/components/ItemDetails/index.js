import React, { Component, Fragment } from 'react';
import './style.css';
import itemImage from '../../assets/red-chair.jpeg';

class ItemDetails extends Component {
  state = {
    isActive: false
  };

  handleClick = () => {
    this.setState({ isActive: !this.state.isActive });
  };
  render() {
    const { item_name, quantity } = this.props.itemDetails;
    const { savings, subtotal } = this.props.pricing;
    const discountedPrice = (subtotal - savings).toFixed(2);

    const item = (
      <div className="item-details">
        <div id="item-picture">
          <img src={itemImage} alt="chair" />
        </div>
        <div>
          <div id="item-name">{item_name}</div>
          <div id="item-qty">
            <div className="item-saving-prices">
              <span className="bold-text">${discountedPrice}</span>
              <span className="bold-text item-subtotal">${subtotal}</span>
            </div>
            <div>Qty: {quantity}</div>
          </div>
        </div>
      </div>
    );

    const itemDetails = this.state.isActive ? (
      <Fragment>
        <div className="Item-Details-Title" onClick={this.handleClick}>
          <h4>Hide item details </h4> <span className="minus" />
        </div>
        {item}
        <hr />
      </Fragment>
    ) : (
      <div onClick={this.handleClick} className="Item-Details-Title">
        <h4>See item details </h4> <span className="add" />
      </div>
    );
    return <div className="Item-Details-Main">{itemDetails}</div>;
  }
}

export default ItemDetails;
