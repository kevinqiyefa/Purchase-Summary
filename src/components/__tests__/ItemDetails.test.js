import React from 'react';
import { shallow } from 'enzyme';
import ItemDetails from '../ItemDetails';
import toJson from 'enzyme-to-json';

describe('Testing ItemDetails Component', () => {
  let wrapped;
  const pricing = {
    subtotal: 102.96,
    savings: 3.85,
    tax: 8.92,
    total: 108.03,
    zip: 85050
  };

  const itemDetails = {
    item_name:
      'Essntials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red',
    quantity: 1
  };

  beforeEach(() => {
    wrapped = shallow(
      <ItemDetails pricing={pricing} itemDetails={itemDetails} />
    );
  });

  it('shows 1 child DIV', () => {
    expect(wrapped.find('.Item-Details-Main').children().length).toEqual(1);
  });

  it('shows state', () => {
    expect(wrapped.state('isActive')).toEqual(false);
  });

  it('shows item details', () => {
    wrapped.find('.Item-Details-Title').simulate('click');
    expect(wrapped.state('isActive')).toEqual(true);
  });

  it('matches snapshot', () => {
    const serialized = toJson(wrapped);
    expect(serialized).toMatchSnapshot();
  });
});
