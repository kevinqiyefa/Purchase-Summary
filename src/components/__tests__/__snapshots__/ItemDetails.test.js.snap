// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ItemDetails Component matches snapshot 1`] = `
<div
  className="Item-Details-Main"
>
  <div
    className="Item-Details-Title"
    onClick={[Function]}
  >
    <h4>
      See item details 
    </h4>
     
    <span
      className="add"
    />
  </div>
</div>
`;
